local_pathfinding/objectives.py

```diff
@@ -24,57 +35,316 @@ def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+
+        # Calculate the uncorrected turn size [0, 2pi]
+        turn_size_bias = math.fabs(global_goal_direction - path_direction)
+
+        # Correct the angle in between [0, pi]
+        if turn_size_bias > math.pi:
+            turn_size_unbias = turn_size_bias - 2 * math.pi
+        else:
+            turn_size_unbias = turn_size_bias
+
+        if turn_size_unbias > large_turn_threshold:
+            return math.degrees(math.fabs(turn_size_unbias))
+
+        return math.degrees(math.fabs(turn_size_unbias))
+
+    def goal_heading_turn_cost(self, s1: ob.SE2StateSpace):
+        """Determine the smallest turn angle between s1-s2 and heading
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            float: the turning angle from s2 to s1 (degrees)
+        """
+        large_turn_threshold = math.pi / 2
+
+        # Calculate the angle from s1-goal line segment from North
+        global_goal_direction = math.atan2(self.goal_x - s1.getX(), self.goal_y - s1.getY())
+
+        # Calculate the uncorrected turn size [0, 2pi]
+        turn_size_bias = math.fabs(global_goal_direction - self.heading_degrees)
+
+        # Correct the angle in between [0, pi]
+        if turn_size_bias > math.pi:
+            turn_size_unbias = turn_size_bias - 2 * math.pi
+        else:
+            turn_size_unbias = turn_size_bias
+
+        if turn_size_unbias > large_turn_threshold:
+            return math.degrees(math.fabs(turn_size_unbias))
+
+        return math.degrees(math.fabs(turn_size_unbias))
+
+    def heading_path_turn_cost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace):
+        """Generates the turning cost between s1-s2 and heading of the sailbot
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            float: The minimum turning angle between s1-s2 and heading  (degrees)
+        """
+        large_turn_threshold = math.pi / 2
+
+        # Calculate the angle of the s1-s2 line segment from North
+        path_direction = math.atan2(s2.getX() - s1.getX(), s2.getY() - s1.getY())
+
+        # Calculate turn size
+        turn_size_bias = math.fabs(path_direction - self.heading_degrees)
+
+        # Correct the angle in between [0, pi]
+        if turn_size_bias > math.pi:
+            turn_size_unbias = turn_size_bias - 2 * math.pi
+        else:
+            turn_size_unbias = turn_size_bias
+
+        if turn_size_bias > large_turn_threshold:
+            return math.degrees(math.fabs(turn_size_unbias))
+
+        return math.degrees(math.fabs(turn_size_unbias))
 
 
 class WindObjective(Objective):
-    def __init__(self, space_information, wind_direction_degrees: float):
+    """Generates a wind objective function"""
```

Comment:
Document attribute `wind_direction` We don't need to include units in variable name, but it should be stated in docstrings.

Reply:
Also I think if you make this radians you don't have to convert anywhere else in `WindObjective`